transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -198,6 +202,17 @@ internal class MessagingClientImpl(
         messageStore.invalidateConversationCache()
     }
 
+    @Throws(IllegalStateException::class)
+    override fun userIsTyping() {
+        val encodedJson = userTypingProvider.encodeRequest(token)
+        if (encodedJson == null) {
+            log.w { "Typing event can be sent only once every $TYPING_INDICATOR_COOL_DOWN_IN_MILLISECOND milliseconds." }
```

Comment:
If we're in this cool-down interval, maybe we should throw an error to let the caller know they were rate limited.

Reply:
There was some back and forth with myself about that, but I could not find good arguments to prefer throwing an exception on top of the logging. In such a case, throwing an exception seems excessive. In any case, if we throw an exception, the UI side doesn't have anything to do with it.
Additionally, in a potential implementation, this function might be invoked whenever an input field changes. There will be a lot of exceptions generated by that.

Reply:
Ok, I think that makes sense to me. My thought was that maybe the customer would want to know when they're being rate limited, but in practical use of this API it probably isn't needed. Like you mention, the most common use would be to just call the API on input field changes and let us just handle the rate limiting to ensure we don't send more than one typing indicator in any 5 second interval.